weaviate/collection/collection_base.py

```diff
@@ -9,12 +9,93 @@
 from weaviate.data.replication import ConsistencyLevel
 from weaviate.exceptions import UnexpectedStatusCodeException, ObjectAlreadyExistsException
 from weaviate.util import _to_beacons, _capitalize_first_letter
-from weaviate.weaviate_classes import CollectionConfigBase, UUID, Metadata
+from weaviate.weaviate_classes import CollectionConfigBase, UUID, Metadata, Tenant
 from weaviate.weaviate_types import UUIDS
 
 
+class _Tenants:
+    """
+    Represents all the CRUD methods available on a collection's multi-tenancy specification within Weaviate. The
+    collection must have been created with multi-tenancy enabled in order to use any of these methods. This class
+    should not be instantiated directly, but is available as a property of the Collection class under
+    the `collection.tenants` class attribute.
+
+    Methods:
+        - `add(tenants: List[Tenant]) -> None`: Add the specified tenants to a collection in Weaviate.
+        - `remove(tenants: List[str]) -> None`: Remove the specified tenants from a collection in Weaviate.
+        - `get() -> List[Tenant]`: Return all tenants currently associated with a collection in Weaviate.
```

Comment:
do you think we need the list of tenants? I think this will get stale quickly and will be autogenerated anyways

> This class should not be instantiated directly, but is available as a property of the Collection class under the `collection.tenants` class attribute.

I think one underscore is enough to show that it is private. We could even do two to make extra clear that you're not supposed to access it